packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2093,6 +2180,8 @@ export class KeyringController extends BaseController<
 
     this.#password = password;
 
+    await this.#updateEncryptedEncryptionKey(password, encryptionKey);
```

Comment:
Possibly related to my comment regarding `encryptedEncryptionKey` rollback, here we are kinda breaking the pattern of updating relevant state data in a single place (`#updateVault` till now).

Though I also understand why it was done, and I don't have yet a clear suggestion to change this - perhaps change the way we use `this.#password` since in envelope mode we don't really have a reason to cache it I guess

Reply:
as mentioned above.

this pattern was not obvious to me and frankly not documented very well.
i've only became aware of it yesterday evening when i realized that the rollback didn't work as expected.
i'll push an update that changes how the encryptedEncryptionKey is handled (similar to password).

Reply:
i think we could add a check in the rollback function that ensures that state is not altered. otherwise issues a warning or something.

Reply:
did update the envelope encryption pattern, please check